### PR TITLE
Homework 1

### PR DESCRIPTION
**Pull request by slavon0089**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -8,9 +8,112 @@
...
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.1</version>
+                <configuration>
+                    <source>11</source>
+                    <target>11</target>
+                </configuration>
```
Comment (obitkin) [inline pom.xml]: Java version duplication. Remove

File: src/main/java/Main.java [hunk 2]
```diff
@@ -1,6 +1,6 @@
 public class Main {
 
     public static void main(String[] args) {
-        System.out.println("Wow");
+
     }
 }
```
Comment (obitkin) [inline src/main/java/Main.java:6]: You can just delete it

File: src/test/java/com/epam/tc/hw1/AbstractTest.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    public void before() {
+        calculator = new Calculator();
+        softAssert = new SoftAssert();
+    }
+
+    @AfterMethod
+    public void after() {
+        calculator = null;
+        softAssert = null;
+    }
```
Comment (obitkin) [inline src/test/java/com/epam/tc/hw1/AbstractTest.java]: Do you really need softAssert in all your classes? i think that SoftAssert should be method variable.

File: src/test/java/com/epam/tc/hw1/DivTest.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        var act = calculator.div(a, b);
+        assertThat(act).isEqualTo(result);
+    }
+
+    @Test(expectedExceptions = NumberFormatException.class)
+    public void negativeDivOnZeroLongTest() {
+        long a = 922337203685772107L;
+        long b = 0;
+        long result = 0;
+        var act = calculator.div(a, b);
```
Comment (obitkin) [inline src/test/java/com/epam/tc/hw1/DivTest.java]: Remove unused `result` and `act`

File: src/test/java/com/epam/tc/hw1/DivTest.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @Test
+    public void negativeDivOnZeroDoubleTest() {
+        double a = 9223.0;
+        double b = 0.0;
+        double result = 0.0;
+        var act = calculator.div(a, b);
+        softAssert.assertEquals(act, result, 0.0001);
+    }
```
Comment (obitkin) [inline src/test/java/com/epam/tc/hw1/DivTest.java]: 1. double(0.0 / doube(0.0) = double(Infinity) != 9223.0
2. softAssert is useless without `assertAll()` method. 
3. Why are you using softAssert here? Why not hard assert?

File: src/test/java/com/epam/tc/hw1/DivTest.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+    @AfterMethod
+    public void after() {
+        calculator = null;
+        softAssert = null;
+    }
```
Comment (obitkin) [inline src/test/java/com/epam/tc/hw1/DivTest.java]: Remove duplicated after method. Already used in AbstractTest

**Review by obitkin — CHANGES_REQUESTED** [review]
- Rename repo to NameSurname
- Add me as collaborator in this repo

File: src/test/java/com/epam/tc/hw1/DivTest.java (hunk 5 above)
Comment (obitkin) [reply src/test/java/com/epam/tc/hw1/DivTest.java]: Sorry, small fix: double(0.0 / doube(0.0) = double(NAN) != 9223.0
double(1.0 / doube(0.0) = double(Infinity) != 9223.0

File: src/test/java/com/epam/tc/hw1/DivTest.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        double result = Double.POSITIVE_INFINITY;;
+        var act = calculator.div(a, b);
+        Assert.assertEquals(act, result, 0.0001);
+    }
+
+    @AfterMethod
+    public void after() {
+        calculator = null;
+
+    }
```
Comment (obitkin) [inline src/test/java/com/epam/tc/hw1/DivTest.java]: Why is it here?

**Review by obitkin — CHANGES_REQUESTED** [review]
Remove `.idea` files from PR

File: src/test/java/com/epam/tc/hw1/DivTest.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+    @Test
+    public void negativeDivOnZeroDoubleTest() {
+        double a = 9223.0;
+        double b = 0.0;
+        double result = Double.POSITIVE_INFINITY;;
+        var act = calculator.div(a, b);
+        Assert.assertEquals(act, result, 0.0001);
+    }
```
Comment (obitkin) [inline src/test/java/com/epam/tc/hw1/DivTest.java]: _9.	Use AssertJ instead of basic TestNg asserts_

**Review by obitkin — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/tc/hw1/DivTest.java (hunk 7 above)
Comment (slavon0089) [reply src/test/java/com/epam/tc/hw1/DivTest.java]: Duplicated by mistake from the parent class. Will be deleted.

**Review by obitkin — APPROVED** [review]
Approved

**Pull request merged**